### DISCUSSION TITLE
How to show router association and how to change router?

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#15206] [General]
First question: How can I determine over which router a device is associated with? And which route it takes (ie which routing devices it traverses to the coordinator)?

Second: I had devices with pretty poor LQIs. My belief always was that Zigbee is a self-healing, resilient as hoc mesh network. As such, I always paired devices next to the coordinator m and brought them to their final destination afterwards. Very certain it would then automatically take the best route.
I found sources claiming best practice is to pair devices at their final destination as they may pair to a different router.
My trust in Zigbee got completely shattered when I found this is actually true: I removed said low LQI decides and repaired at their intended destination and lo and behold, some LQIs improved from 20 to over 100.

Now 2nd question: Can I change this “association” somehow without repairing devices?

**Comment by rawnsley** [comment]
I've reached a similar place to you and my understanding is this: the "Map" feature in zigbee2mqtt will show you the routes within your zigbee network, including which router the endpoint devices are currently connected to.

All of the devices capable of routing are (not surprisingly) able to maintain multiple connections with other routers and consequently "self-heal". All of the "endpoint" devices I've got on my network (air monitors, buttons, etc...) only connect to a single router at any one time, which makes sense from an energy saving point of view. Some of them will change router if a better option is available, although how long this takes is a mystery (at least hours, possibly days in many cases). Some of them never change unless prompted, which is apparently a deviation from spec common to several brands.

My faith in Zigbee has also been sorely tested by this behaviour and I don't know if it is the protocol or dodgy implementations that are to blame. If it is the latter, then surely the certifications for those manufacturers should be rescinded?

Having reached the acceptance stage of grief with Zigbee, I have started to sift through my network map and find poorly paired devices. zigbee2mqtt lets you open up device paring for specific routers through this combo box:

<img width="256" alt="Screenshot 2023-11-26 at 10 06 50" src="https://github.com/Koenkk/zigbee2mqtt/assets/303516/f7db180f-f2a4-4748-b458-876bec9da1de">

Activate it only for the router you want your endpoint device to pair with, start the routing process on the device (typically a few seconds press of the reset button until something starts flashing) and then it *should* connect through that route point. This usually works, but I have several router-device combinations that refuse to re-pair in this way. Could be a mismatch of zigbee versions or implementation gaps between devices? 

Frankly I shouldn't have to care about this stuff and it seriously undermines the "it-just-works" promises that zigbee makes. I'm sure people will say I'm "holding it wrong" or that I should switch to WiFi (seriously, people say that with a straight face), but this technology is just not ready for civilians.

**Comment by riesi77** [comment]
I experience the same behaviour and explained it under https://github.com/Koenkk/zigbee2mqtt/discussions/19798 from a bit of a different point of view.
Which coordinators do you use? Could this be dependent on them?

**Reply by rawnsley** [reply]
I use a [Sonoff dongle](SONOFF Universal Zigbee 3.0 USB Dongle Plus Gateway). It's possible that the coordinator is a factor; I don't know enough about the protocols involved to say where the decisions and negotiations occur.
